RFC 68 - improve the new project user experience

brylie opened the pull request:
[Rendered](https://github.com/brylie/rfcs/blob/0068-new-project-user-experience/text/068-improve-new-project-user-experience.md)

brylie commented:
For shared reference, this pull request was inspired by watching [two new developers trying out the Wagtail installation process and fresh Wagtail Admin UI](https://www.youtube.com/watch?v=L8YtfFmd-0Q). They admit rushing in to try things out without reading the tutorial, but I suspect they are not alone in the desire to quickly try out the Wagtail Admin UI before diving into code tutorials.

gasman commented:
Thanks @brylie - this is definitely something that's overdue for some new UX consideration...

I think making the initial project state into something immediately usable (so that 'add page' isn't just creating a tree of HomePages) is a good move, but we should be careful not to introduce any new roadblocks for when people move away from the default structure and start building their own thing. The trouble with making HomePage into something more general purpose is that a non-trivial site probably _does_ want a distinct page type for the home page, and once you've built a tree of general-purpose pages it's quite hard to retrospectively turn the homepage into a more specific type (while still keeping the child pages and Site record intact).

Maybe the way to go is to start out with two page types, HomePage and ContentPage, with a `subpage_types = [ContentPage]` rule on HomePage to stop people creating new homepages as children. (Or, perhaps better, [introduce a `RootPage` proxy model for the tree root](https://github.com/wagtail/wagtail/pull/1736#issuecomment-160944662) and set HomePage to `parent_page_types = [RootPage]` - that way we're not adding an extra hoop to jump through when developers bring in a second child page type.) At that point, we probably need to either give ContentPage its own app, or rename the 'home' app to something else...

Also, I'd like to push for the `body` field to be a StreamField rather than a RichTextField. (Or maybe for ContentPage, an RichTextField `intro` and a StreamField `body`?) I think we're already in a situation (partly prompted by the tutorial) where people hang on to RichTextField for complex content long past the point where they should have migrated to StreamField, and steering people towards the 'better' solution from the start would be a good move.

brylie commented:
The `RootPage` proxy model could be useful for other deployments where developers want specific pages to only be at the root level, such as a blog index page.

tombola commented:
+1 for `HomePage` / `ContentPage` so that for their first site a new developer is not left with a `DemoPage` page type or something similar in their codebase (and content in database) that they might then find it difficult to move away from.

brylie commented:
@gasman and @tombola, I've updated the RFC to incorporate most of the suggestions from this thread.

- added `RootPage` proxy model
- `HomePage` with `StreamField -- single instance directly under `RootPage`
- `ContentPage`with `StreamField`
